flyteplugins/go/tasks/plugins/k8s/spark/spark_test.go

```diff
@@ -535,21 +587,20 @@ func TestBuildResourceSpark(t *testing.T) {
 	assert.Equal(t, tolDriverDefault.Effect, corev1.TaintEffect("NoSchedule"))
 
 	assert.Equal(t, 2, len(sparkApp.Spec.Executor.Tolerations))
-	assert.Equal(t, 1, len(sparkApp.Spec.Executor.NodeSelector))
-	assert.Equal(t, interruptibleNodeSelector, sparkApp.Spec.Executor.NodeSelector)
-
-	tolExecDefault := sparkApp.Spec.Executor.Tolerations[0]
-	assert.Equal(t, tolExecDefault.Key, "x/flyte")
-	assert.Equal(t, tolExecDefault.Value, "default")
-	assert.Equal(t, tolExecDefault.Operator, corev1.TolerationOperator("Equal"))
-	assert.Equal(t, tolExecDefault.Effect, corev1.TaintEffect("NoSchedule"))
+	assert.Equal(t, 2, len(sparkApp.Spec.Executor.NodeSelector))
```

Comment:
With this change I'm now expecting BOTH default and interruptible node selectors to be applied to the executor pod. This is consistent with how defaults and interruptible works for other plugins. @hamersaw @jeevb do you anticipate problems with this change?

Reply:
My understanding is that the executor is more important than the workers and as such has historically been scheduled on non-interruptible. This should apply to all driver-worker plugins (e.g. spark, ray). If we do want to go down this path, we should at least make it configurable. This behavior could be undesirable to users that use these plugins at scale.

Reply:
Ah, just to clarify here, the driver will be treated as non-interruptible and get the non-interruptible selectors *in addition* to the default ones and the default tolerations, while the executor pods will honor the task's interruptible param, and if interruptible will get the interruptible selectors *in addition* to the default ones and the interruptible tolerations in addition to the default ones.

Previously the spark plugin **[replaced](https://github.com/flyteorg/flyte/pull/4183/files#diff-5225038bf6b0b87842d5e795c8485ba63d597946970cc180637955e8baf0e38bL249)** the executor's node selector if interruptible vs. appending. This may have been a bug/oversight

Reply:
Ah. For some reason I read `executor` as `driver`, when an `executor` is just a worker. My mistake! 🤦